### DISCUSSION TITLE
Content Security Policy (CSP) support for Simple Light Box

### DISCUSSION THREAD

**Discussion by ricky176** [andreknieriem/simplelightbox#326] [General]
Does simplelightbox support CSP?

If not is there any plan for CSP support in future.

**Comment by Mtillmann** [comment]
Hi, what exactly do you mean? CSP is implemented on the document level of by passing http headers. Simplelightbox is a clientside lib

**Reply by ricky176** [reply]
After implementing CSP during user interaction on image lightbox is adding some inline  style to the DOM element which is blocked by CSP. 
![image](https://github.com/andreknieriem/simplelightbox/assets/47901612/2024feee-7113-4c40-a0b9-96cb11ff0cc8)

![image](https://github.com/andreknieriem/simplelightbox/assets/47901612/6be0e55a-f634-49bc-9df8-1786de942470)

Please. let me know if I am able to explain you properly.

**Comment by Mtillmann** [comment]
Thanks for Info! Are you loading simplelightbox from a cdn host? It's not clear from the screenshots?

**Reply by ricky176** [reply]
> Thanks for Info! Are you loading simplelightbox from a cdn host? It's not clear from the screenshots?

Yes its loaded from cdn host.

**Reply by Mtillmann** [reply]
Please look into how to set up CSP rules to allow scripts from the CDN host you're using. This is not an issue directly related to simplelightbox I believe.

**Reply by ricky176** [reply]
> Please look into how to set up CSP rules to allow scripts from the CDN host you're using. This is not an issue directly related to simplelightbox I believe.

![image](https://github.com/andreknieriem/simplelightbox/assets/47901612/471ba036-69b9-41c6-914c-2e22a109cae8)


You can check in the above screen shot the cdn from where light box is loaded is whitelisted in earlier screenshot of chrome console.


The issue is with lightbox trying to add some style when user interacts which is blocked by csp.

The lightbox js n css both are loaded and available during page load as the cdn is whitelisted.
